synapse/handlers/read_marker.py

```diff
@@ -47,9 +47,11 @@ async def received_client_read_marker(self, room_id, user_id, event_id):
 
             if existing_read_marker:
                 # Only update if the new marker is ahead in the stream
-                should_update = await self.store.is_event_after(
-                    event_id, existing_read_marker["event_id"]
+                token1 = await self.store.get_event_ordering(event_id)
+                token2 = await self.store.get_event_ordering(
+                    existing_read_marker["event_id"]
                 )
+                should_update = token1 > token2
```

Comment:
attr objects by default allow comparisons I'm assuming? (I don't see anything special in `RoomStreamToken` to support this.)

Reply:
Argh, https://github.com/matrix-org/synapse/pull/8439 removes this as `RoomStreamTokens` should *not* be comparable (as they'll be vector clocks).